involute_gears.scad

```diff
@@ -42,488 +42,538 @@ pi=3.1415926535897932384626433832795;
-						triangles=[[0,2,1],[0,3,2],[0,4,3],[0,1,5],[1,2,5],[2,3,5],[3,4,5],[0,5,4]]);
-				}
-			}
-		}
-	}
+//  echo ("involute_bevel_gear_tooth",
+//      back_cone_radius,
+//      root_radius,
+//      base_radius,
+//      outer_radius,
+//      pitch_apex,
+//      cone_distance,
+//      half_thick_angle);
+
+    min_radius = max (base_radius*2,root_radius*2);
+
+    pitch_point =
+        involute (
+            base_radius*2,
+            involute_intersect_angle (base_radius*2, back_cone_radius*2));
+    pitch_angle = atan2 (pitch_point[1], pitch_point[0]);
+    centre_angle = pitch_angle + half_thick_angle;
+
+    start_angle = involute_intersect_angle (base_radius*2, min_radius);
+    stop_angle = involute_intersect_angle (base_radius*2, outer_radius*2);
+
+    res=(involute_facets!=0)?involute_facets:($fn==0)?5:$fn/4;
+
+    translate ([0,0,pitch_apex])
+    rotate ([0,-atan(back_cone_radius/cone_distance),0])
+    translate ([-back_cone_radius*2,0,-cone_distance*2])
+    union ()
+    {
+        for (i=[1:res])
+        {
+            let (
+                point1=
+                    involute (base_radius*2,start_angle+(stop_angle - start_angle)*(i-1)/res),
+                point2=
+                    involute (base_radius*2,start_angle+(stop_angle - start_angle)*(i)/res))
+            {
+                let (
+                    side1_point1 = rotate_point (centre_angle, point1),
+                    side1_point2 = rotate_point (centre_angle, point2),
+                    side2_point1 = mirror_point (rotate_point (centre_angle, point1)),
+                    side2_point2 = mirror_point (rotate_point (centre_angle, point2)))
+                {
+                    polyhedron (
+                        points=[
+                            [back_cone_radius*2+0.1,0,cone_distance*2],
+                            [side1_point1[0],side1_point1[1],0],
+                            [side1_point2[0],side1_point2[1],0],
+                            [side2_point2[0],side2_point2[1],0],
+                            [side2_point1[0],side2_point1[1],0],
+                            [0.1,0,0]],
+                        triangles=[[0,2,1],[0,3,2],[0,4,3],[0,1,5],[1,2,5],[2,3,5],[3,4,5],[0,5,4]]);
+                }
+            }
+        }
+    }
 }
 
 module gear (
-	number_of_teeth=15,
-	circular_pitch=undef, diametral_pitch=undef,
-	pressure_angle=28,
-	clearance = undef,
-	gear_thickness=5,
-	rim_thickness=undef,
-	rim_width=undef,
-	hub_thickness=undef,
-	hub_diameter=undef,
-	bore_diameter=undef,
-	circles=0,
-	backlash=0,
-	twist=0,
-	involute_facets=0,
-	flat=false)
+    number_of_teeth=15,
+    circular_pitch=undef, diametral_pitch=undef,
+    pressure_angle=28,
+    clearance = undef,
+    gear_thickness=5,
+    rim_thickness=undef,
+    rim_width=undef,
+    hub_thickness=undef,
+    hub_diameter=undef,
+    spokes=0,
+    spoke_width=undef,
+    spoke_thickness=undef,
+    spoke_square=false,
+    z_symmetry=false,
```

Comment:
Does this flag do anything?

Reply:
acccedaently left this in. I was going to implement the z_symetry option to center all gear parts on the XY plane. It is useful if you do not want offset hubs like the code produces now. I took out the supporting code so that I could check in incremental steps rather than doing everything in one PR. It should not have been left in.